### PR TITLE
Implemented Grid class for the simulation

### PR DESCRIPTION
**Pull request by Ronenii**
- Added the class Grid.java which is responsible for managing the grid of the entity instances.
- Implemented movement logic for instances.
- Implemented basic grid functions in World.java

### CODE REVIEW

File: Engine/src/simulation/objects/entity/EntityInstance.java [hunk 1]
```diff
@@ -3,18 +3,39 @@
...
 import java.util.Map;
 
 public class EntityInstance implements Serializable {
+    private final Entity contextEntity;
     private final Map<String, Property> properties;
     private boolean isAlive;
-    public int xGridCoordinate;
-    public int yGridCoordinate;
+    private int xGridCoordinate;
+    private int yGridCoordinate;
```
Comment (RoyToledano) [inline Engine/src/simulation/objects/entity/EntityInstance.java]: I think both should be public.
If I remember correctly, Aviad or Guy (or both) said that getters & setters without any other logic considered bad practice.

File: Engine/src/simulation/properties/action/impl/proximity/ProximityAction.java [hunk 2]
```diff
@@ -24,13 +24,13 @@ public Object getValue() {
         return null;
     }
 
-    public void Invoke(EntityInstance firstEntityInstance, EntityInstance secondEntityInstance, EntityInstance[][] grid, int lastChangeTickCount) {
+    public void invoke(EntityInstance firstEntityInstance, EntityInstance secondEntityInstance, EntityInstance[][] grid, int lastChangeTickCount) {
```
Comment (RoyToledano) [inline Engine/src/simulation/properties/action/impl/proximity/ProximityAction.java]: If we implement a 'Grid' object we need to send this object to proximity's invoke, and not the two dimensional array.

**Review by RoyToledano — CHANGES_REQUESTED** [review]
Comments added

File: Engine/src/simulation/properties/action/impl/proximity/ProximityAction.java (hunk 2 above)
Comment (Ronenii) [reply Engine/src/simulation/properties/action/impl/proximity/ProximityAction.java]: And done!

File: Engine/src/simulation/objects/entity/EntityInstance.java (hunk 1 above)
Comment (Ronenii) [reply Engine/src/simulation/objects/entity/EntityInstance.java]: Done

**Review by RoyToledano — APPROVED** [review]
amazing

**Pull request merged**